### DISCUSSION TITLE
Convert boolean in float

### DISCUSSION THREAD

**Discussion by Simone2021space** [sympy/sympy#22289] [Q&A] [answered]
Hi team,
I need to evaluate the following expression in sympy:
(z < 0.1) * 1.0

where z is a symbol, so if z is less than 0.1 the expression is 1, otherwise it is 0.
In python this is possible, but with sympy I get the following error:

"TypeError: unsupported operand type(s) for *: 'StrictLessThan' and 'float' " 
Any idea to solve this?

**Comment by redeboer** [comment, marked as answer]
Sounds like you are looking for [`Piecewise`](https://docs.sympy.org/latest/modules/functions/elementary.html#sympy.functions.elementary.piecewise.Piecewise):
```python
>>> import sympy as sp
>>> z = sp.Symbol("z")
>>> expr = sp.Piecewise(
...     (z, z < 0.1),
...     (1.0, True),
... )
>>> expr.subs(z, .05)
0.05
>>> expr.subs(z, 2)
1.0
```

**Comment by oscarbenjamin** [comment]
There is an issue about adding an indicator function #19481
